CatSwipe/Views/CollectionPage.xaml.cs

```diff
@@ -42,4 +57,62 @@ private async Task LoadLikedCatsAsync()
             await DisplayAlert("Error", $"Failed to load your cat collection: {ex.Message}", "OK");
         }
     }
+
+    private async void OnCatTapped(object? sender, EventArgs e)
+    {
+        if (sender is not Border border || border.BindingContext is not Cat cat)
+            return;
+
+        try
+        {
+            await ShareCatAsync(cat);
+        }
+        catch (Exception ex)
+        {
+            await DisplayAlert("Error", $"Unable to share cat: {ex.Message}", "OK");
+        }
+    }
+
+    private async Task ShareCatAsync(Cat cat)
+    {
+        // Create a temporary file to download the image
+        var tempPath = Path.Combine(FileSystem.CacheDirectory, $"cat_{cat.Id}.jpg");
+
+        try
+        {
+            // Show loading indicator
+            IsSharing = true;
+
+            // Download the cat image
+            using var response = await _httpClient.GetAsync(cat.ImageUrl);
+            response.EnsureSuccessStatusCode();
+
+            // Save image to temporary file
+            using var fileStream = File.Create(tempPath);
+            await response.Content.CopyToAsync(fileStream);
+
+            // Share the image
+            await Share.Default.RequestAsync(new ShareFileRequest
+            {
+                Title = $"Check out this {cat.Breed ?? "adorable cat"}!",
+                File = new ShareFile(tempPath)
+            });
```

Comment:
I was testing this on Android, and the share popup opens. But I don't see the image being shared.

Can you search the web and/or docs to see what might be wrong here?